File: P120848.py
def solution(n):
    dp = [0] * 11
    dp[1] = 1
    for i in range(2, 11):
        dp[i] = i*dp[i-1]
    for i in range(1, 11):
        if dp[i] == n:
            answer = i
            break
        elif dp[i-1] < n < dp[i]:
            answer = i-1
            break
    return answer

File: P120848.java
public class P120848 {
    public int solution(int n) {
        int answer = 0;
        int[] factorial = new int[11];
        factorial[1] = 1;
        for (int i=2; i<11; i++) factorial[i] = i*factorial[i-1];
        for (int i=10; i>0; i--) {
            if (n >= factorial[i]) {
                answer = i;
                break;
            }
        }
        return answer;
    }
}
